### PR TITLE
merge conflicts resolved for Snake example

### PR DESCRIPTION
**Pull request by pjforde1978**

### CODE REVIEW

File: app/controllers/demos_controller.rb [hunk 1]
```diff
@@ -25,6 +25,27 @@ def show
...
+        session[:start_x] = 50
+        session[:start_y] = 50
+        session[:food] = []
+        session[:alive] = true
+      end
+      @grid_x = session[:grid_x] * 10 + 1
+      @grid_y = session[:grid_y] * 10 + 1
+    end
+  end
+
```
Comment (hopsoft) [inline app/controllers/demos_controller.rb]: Note that this isn't necessary as `setup_snake` handles the init for the snake demo.

Comment (pjforde1978) [reply app/controllers/demos_controller.rb]: I have some additional changes coming on this PR, so hold off.

I'll investigate setup_snake.

**Comment by pjforde1978** [comment]
This should be better in every way, but I wanted to draw some attention to a problem I'm becoming increasingly aware of, which is that often you see several second's delay before the websocket connection is fully mounted and ready for messages.

As I noted on Workflowy, LiveView provides developers with a solid mechanism for handling this:

https://hexdocs.pm/phoenix_live_view/Phoenix.LiveView.html#module-loading-state-and-errors

Chris throws some connection state flags as classes on the body element. This is really smart because instead of the nasty-ass

```JS
this.StimulusReflex.subscription.consumer.connection.webSocket.addEventListener(
      'open',
      this.enableStart.bind(this)
    )
```

...I just had to do, I could accomplish the same thing with a few CSS rules on the elements in question. Would you be interested in seeing a PR that adds similar functionality to StimulusReflex?

**Comment by hopsoft** [comment]
Let me look into that a bit more.

**Comment by pjforde1978** [comment]
Awesome! Excited to see this live. If you were looking for a way to test SR on a free vs Hobby dyno, look no further. :grin:

**Comment by pjforde1978** [comment]
Hey, I just noticed that you moved the `snake_controller.rb` to `app/controllers/concerns`.

I understand why you didn't want to include the Scss, but I don't think this was the best way to handle this. While it's true that it's is only being used as a demonstration in the expo application, someone seeing that it's in the concerns folder is just going to be (rightfully) confused.

**Pull request merged**